driver/ni/ni.cpp

```diff
@@ -92,65 +519,56 @@ void ni::Source::getIndexKeys(){
 ni::Source::Source(
     TaskHandle task_handle,
     const std::shared_ptr<task::Context> &ctx,
-    const synnax::Task task): task_handle(task_handle), ctx(ctx), task(task){
+    const synnax::Task task) : task_handle(task_handle), ctx(ctx), task(task) {
 }
 
-void ni::Source::parseConfig(config::Parser &parser){
-    
-    LOG(INFO) << "[NI Source] Parsing Configuration for task " << this->reader_config.task_name;
+void ni::Source::parse_config(config::Parser &parser) {
     // Get Acquisition Rate and Stream Rates
-    this->reader_config.sample_rate = parser.required<uint64_t>("sample_rate");
-    this->reader_config.stream_rate = parser.required<uint64_t>("stream_rate");
+    this->reader_config.sample_rate.value = parser.required<uint64_t>("sample_rate");
```

Comment:
You're doing a type narrowing here from an unsigned long to a float. Same with the next line. Should we be able to support sample rates in the range 0 < sr < 1?

Reply:
I wouldn't think so. I've never seen a device that supports that natively and would be hard pressed to imagine someone would ever want to sample date at such a low rate.